pkgs/pkgs-lib/formats.nix

```diff
@@ -92,6 +92,10 @@ rec {
     # Alternative to listsAsDuplicateKeys, converts list to non-list
     # listToValue :: [IniAtom] -> IniAtom
     listToValue ? null,
+    # Allows atoms directly at the root
+    globalSection ? false,
+    # Appended to generated text
+    extraConfig ? null, # null or string
```

Comment:
I think having this as an argument to the generator isn't the best, but just a quick hack so I can get feedback.

My preferred solution would just be to remove `services.gitea.extraConfig` and not have to deal with this at all. For Gitea since the config format is plain ini, `extraConfig` has very little value since Nix can already represent everything that `extraConfig` can.

I do see the value in not breaking existing configs though, so if we want to keep the `extraConfig` option, I think this would be better implemented by adding a function to the format. Something like `generateText` which would do what `generate` does but without creating a file, just returning a string.  

Basically:
```nix
{
  generateText = value: ...;
  generate = name: value: pkgs.writeText name (generateText value);
}
```

Seeing as formats where defined with an RFC, I wanted to get feedback before doing that.
Adding `generateText` is far from as major as introducing formats, so I didn't think I'd create an RFC just yet. And this is my first real Nix(OS) contribution so that'd be a bit much :smile:

Reply:
In order to remove `extraConfig`, we need to deprecate it at least for one release cycle. I don't recall if this has been done or not.

Under the assumption, we cannot, I feel like having this `extraConfig` is a bit much, but I would defer to @roberth @infinisil who may have better opinions on this.

Reply:
Yeah definitely understand we'd want to deprecate it first. So I think the `generateText` is the best realistic approach and would appreciate feedback on it.  
@infinisil is probably one the best people to do so, thanks for mentioning them!

Reply:
I'd prefer to keep the `generate` name, so as not to change the "type" of generators.

By keeping things "regular", it can be handled better by code.

I don't foresee any real problems in the most complicated use case I'm aware of, below, but it does illustrate that sticking to the standard is safer.

https://github.com/hercules-ci/flake.parts-website/blob/fa9f76d7168c0ad02abec0fcdbf2ce16d23681d3/render/render-module.nix#L24-L40

Reply:
Maybe my comment was unclear, but `generateText` would be in addition to `generate`, to keep the "type" backwards compatible.  
See https://github.com/NixOS/nixpkgs/pull/206179/commits/55399f49ca24dead0bde1a87e860735993c6d65f for an implementation of the idea, hopefully an diff is worth a 1000 words!

Reply:
> backwards compatible.

`pkgs.formats` won't fit in a `mapAttrs (name: { type, generate }: foo)` anymore, or a `{ type, generate, generateText }:` for that matter.

I think I'm ok with that, but I would like @infinisil to review the idea.

Reply:
I don't think `generateText` is needed, see https://github.com/NixOS/nixpkgs/pull/206179#discussion_r1055695593